From: Sascha Hauer <s.hauer@pengutronix.de>
Date: Mon, 29 Aug 2016 17:31:14 +0200
Subject: [PATCH] blspec: Handle nfs:// pathes from boot

Fixes: cb47dde boot: Call blspec_scan_directory() only on strings containing an absolute path

This commit introduced a check if the path contains a '/' at the
beginning. For booting a bootloader spec entry from NFS we have to
test the path for starting with "nfs://" aswell.

Signed-off-by: Sascha Hauer <s.hauer@pengutronix.de>
---
 common/boot.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/common/boot.c b/common/boot.c
index 573bebde9e2e..2b0e62c33bbd 100644
--- a/common/boot.c
+++ b/common/boot.c
@@ -251,6 +251,7 @@ static int bootscript_scan_path(struct bootentries *bootentries, const char *pat
  * - a cdev name
  * - a full path of a directory containing bootloader spec entries
  * - a full path of a directory containing bootscripts
+ * - a nfs:// path
  *
  * Returns the number of entries found or a negative error code.
  */
@@ -264,7 +265,7 @@ int bootentry_create_from_name(struct bootentries *bootentries,
 		if (ret > 0)
 			found += ret;
 
-		if (*name == '/') {
+		if (*name == '/' || !strncmp(name, "nfs://", 6)) {
 			ret = blspec_scan_directory(bootentries, name);
 			if (ret > 0)
 				found += ret;
